arch/rv32i/src/syscall.rs

```diff
@@ -40,6 +44,56 @@ const R_A2: usize = 11;
 const R_A3: usize = 12;
 const R_A4: usize = 13;
 
+const MAJOR_VER: u32 = 1;
+const MINOR_VER: u32 = size_of::<Riscv32iStoredState>() as u32;
+const TAG: [u8; 4] = [b'r', b'v', b'5', b'i'];
+const METADATA_LEN: usize = 3;
+
+const U32_SZ: usize = size_of::<u32>();
+fn u32_byte_range(index: usize) -> Range<usize> {
+    index * U32_SZ..(index + 1) * U32_SZ
+}
+
+fn u32_from_u8_slice(slice: &[u8], index: usize) -> Result<u32, ErrorCode> {
+    let range = u32_byte_range(index);
+    if range.end < slice.len() {
+        Ok(u32::from_le_bytes(
+            slice[range].try_into().or(Err(ErrorCode::FAIL))?,
```

Comment:
All of the index operations make me wish Rust didn't support that syntax, because we have to rely on the compiler to figure out that there are enough checks to ensure a panic can't happen and to elide the panic handling code.

I personally would rather see `.get()` used rather than `[]` with the `if()` statement (as it is here: https://github.com/tock/tock/blob/master/libraries/tock-tbf/src/types.rs#L426), but this is a little bit of my pet peeve and not something we are rigorous about in the kernel. I think the code gets a little simpler, but I won't hold up this PR on this issue.